src/charm.py

```diff
@@ -43,6 +45,66 @@ def __init__(self, *args: typing.Any) -> None:
         super().__init__(*args)
         self.framework.observe(self.on.config_changed, self.reconciliation)
         self.framework.observe(self.on.spring_boot_app_pebble_ready, self.reconciliation)
+        self.database: DatabaseRequires = self._setup_database_requirer("mysql", "spring-boot")
+
+    def _setup_database_requirer(self, relation_name: str, database_name: str) -> DatabaseRequires:
+        """Set up a DatabaseRequires instance.
+
+        The DatabaseRequires instance is an interface between the charm and various data providers.
+        It handles those relations and emit events to help us abstract these integrations.
+
+        Args:
+            relation_name: Name of the data relation
+            database_name: Name of the database (can be overwritten by the provider)
+
+        Returns:
+            DatabaseRequires object produced by the data_platform_libs.v0.data_interfaces library
+        """
+        database_requirer = DatabaseRequires(
+            self,
+            relation_name=relation_name,
+            database_name=database_name,
+        )
+        self.framework.observe(database_requirer.on.database_created, self.reconciliation)
+        self.framework.observe(self.on[relation_name].relation_broken, self.reconciliation)
+        return database_requirer
+
+    def _datasource(self) -> dict[str, str]:
+        """Compute datasource dict and return it.
+
+        Returns:
+            Dict containing details about the data provider integration
+        """
+        # Default results when something went wrong
+        default = {
+            "username": "",
+            "password": "",
+            "url": "",
+        }
```

Comment:
could we define a defined structure for what the datasource information may be needed? for instance a `NamedTuple`? :)

Reply:
I think that the closest to what you're asking would be to use a dataclass and then return the dict equivalent of it at the end. But that would add a lot just to define the structure of the dict inside this function. 
 
I don't think that it's a good idea to use something else than a dict when we're building the application_config of the charm since a lot of values could be added to it in the future